sympl/_core/wrappers.py

```diff
@@ -0,0 +1,149 @@
+            was cached before new output is computed.
+        """
+        self._prognostic = prognostic
+        self._update_timedelta = update_timedelta
+        self._cached_output = None
+        self._last_update_time = None
+
+    def __call__(self, state, **kwargs):
+        if ((self._last_update_time is None) or
+                (state['time'] >= self._last_update_time +
+                 self._update_timedelta)):
+            self._cached_output = self._prognostic(state, **kwargs)
+            self._last_update_time = state['time']
+        return self._cached_output
+
+    def __getattr__(self, item):
+        return getattr(self._prognostic, item)
+
+
+class TendencyInDiagnosticsWrapper(object):
+    """
+    Wraps a prognostic object so that when it is called, it returns all
+    tendencies in its diagnostics.
+
+    Example
+    -------
+    This how the wrapper should be used on a fictional Prognostic class
+    called RRTMRadiation.
+
+    >>> prognostic = TendencyInDiagnosticsWrapper(RRTMRadiation(), 'radiation')
+    """
+
+    def __init__(self, prognostic, label):
+        """
+        Initialize the Delayed object.
+
+        Args
+        ----
+        prognostic : Prognostic
+            The object to be wrapped
+        label : str
+            Label describing what the tendencies are due to, to be
+            put in the diagnostic quantity names.
+        """
+        self._prognostic = prognostic
+        self._tendency_label = label
+        self._tendency_diagnostic_properties = {}
+        for quantity_name, properties in prognostic.tendency_properties.items():
+            diagnostic_name = 'tendency_of_{}_due_to_{}'.format(quantity_name, label)
+            self._tendency_diagnostic_properties[diagnostic_name] = properties
+
+    @property
+    def inputs(self):
+        return list(self.diagnostic_properties.keys())
+
+    @property
+    def input_properties(self):
+        return_dict = self._prognostic.input_properties.copy()
+        return_dict.update(self._tendency_diagnostic_properties)
+        return return_dict
+
+    @property
+    def diagnostics(self):
+        return list(self.diagnostic_properties.keys())
+
+    @property
+    def diagnostic_properties(self):
+        return_dict = self._prognostic.diagnostic_properties.copy()
+        return_dict.update(self._tendency_diagnostic_properties)
+        return return_dict
+
+    def __call__(self, state, **kwargs):
+        tendencies, diagnostics = self._prognostic(state, **kwargs)
+        for quantity_name in tendencies.keys():
+            diagnostic_name = 'tendency_of_{}_due_to_{}'.format(
+                quantity_name, self._tendency_label)
+            diagnostics[diagnostic_name] = tendencies[quantity_name]
+        return tendencies, diagnostics
+
+    def __getattr__(self, item):
+        return getattr(self._prognostic, item)
+
+
+class ImplicitPrognosticWrapper(ImplicitPrognostic):
```

Comment:
call this TimeDifferenced or something similar?

Reply:
Renamed it to TimeDifferencingWrapper.